game-core/src/main/java/games/strategy/triplea/ui/panels/map/MapPanel.java

```diff
@@ -746,13 +740,15 @@ public void paint(final Graphics g) {
     updateUndrawnTiles(undrawnTiles, 513);
     updateUndrawnTiles(undrawnTiles, 767);
     clearPendingDrawOperations();
+    // make sure we use the same data for the entire paint
+    final GameData data = gameData;
```

Comment:
Kinda odd as `GameData` is quite mutable. I wonder if this would be more easily communicated if we extracted the logic that needs `GameData` to a method. Thoughts?

Reply:
IIRC the reason this variable was introduced is because we got NPEs on shutdown, when the gameData field was set to null mid-render.
This prevents this issue.

Reply:
If we sent the `gameData` reference as a method parameter, there would be no way for anything outside of the method to mutate that reference (value). We would surely guard against that scenario. Given this is also a long method, breaking up the dependencies on `gameData` could be good as well.

IMO that is a good option. If not, then we probably need more commentary/documentation to state we are defending against the `gameData` reference from not only being changed, but also potentially nulled out as part of a shutdown event.